dialog_encdec.py

```diff
@@ -743,23 +758,29 @@ def __init__(self, state):
 
         logger.debug("Build encoder")
         self.h, self.hs = self.encoder.build_encoder(training_x, xmask=training_hs_mask)
-        
+        #target_probs, target_probs_full_matrix = self.decoder.build_decoder(hs, training_x, xmask=training_hs_mask, y=training_y, mode=Decoder.EVALUATION)
+
         logger.debug("Build decoder (NCE)")
         contrastive_cost, self.hd_nce = self.decoder.build_decoder(self.hs, training_x, y_neg=self.y_neg, y=training_y, xmask=training_hs_mask, mode=Decoder.NCE)
         
         logger.debug("Build decoder (EVAL)")
-        target_probs, self.hd, self.decoder_states = self.decoder.build_decoder(self.hs, training_x, xmask=training_hs_mask, y=training_y, mode=Decoder.EVALUATION)
+        target_probs, self.hd, self.decoder_states, target_probs_full_matrix = self.decoder.build_decoder(self.hs, training_x, xmask=training_hs_mask, y=training_y, mode=Decoder.EVALUATION)
+
```

Comment:
Ok i think that theano can free the memory if we do not return it.. have you noticed any differences?